app.py

```diff
@@ -102,15 +130,41 @@ def post_by_id(id_: int):
 	update message by id (if the user is the poster)
 	delete message by id (if the user is the poster)
 	"""
-	...
+	if not session['is_banned']:
+		if request.method == 'GET':
+			get_posts_by_id: list[Posts] = Posts.query.filter_by(id=id_).all()
+			all_dict_posts_by_id: list[dict] = [post.get_dict_post() for post in get_posts_by_id]
+			return make_response(jsonify(all_dict_posts_by_id), 200)
+		elif request.method == 'PUT':
+			posts_by_id: list[Posts] = Posts.query.filter_by(id=id_).all()
+			if session['user'] == posts_by_id[0].user:
+				posts_by_id[0].body = request.form['body']
+				db.session.commit()
+				return make_response(jsonify({'task': 'put', 'status': 'success'}), 200)
+			return make_response(jsonify({'task': 'put', 'status': 'failed'}), 401)
+		elif request.method == 'DELETE':
+			posts_by_id: list[Posts] = Posts.query.filter_by(id=id_).all()
+			if session['user'] == posts_by_id[0].user:
+				db.session.delete(posts_by_id[0])
+				db.session.commit()
+				return make_response(jsonify({'task': 'delete', 'status': 'success'}), 200)
+			return make_response(jsonify({'task': 'delete', 'status': 'failed'}), 401)
+	return make_response(jsonify({'task': 'post', 'status': 'failed', 'reason': 'user is banned'}), 403)
 
 
 @app.route('/users/<int:id_>', methods=['PUT'])
 def user_by_id(id_: int):
 	"""
 	update is_banned (if the user is admin)
 	"""
-	...
```

Comment:
add decorator that checks if the user logged in, if not return 403, you can make 2 decorators, one for user and one for admin and user them in every enpoint that requires it